### PR TITLE
Fix and simplify module type determination upon deployment

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

File: spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java [hunk 1]
```diff
@@ -286,37 +284,19 @@ private ModuleDefinition postProcessLibraryProperties(ModuleDefinition module) {
...
-				// stream begins with a named channel in the source position
-				type = ArtifactType.processor;
-			}
-			else { 
-				type = ArtifactType.source;
-			}
-		}
-		else {
-			type = ArtifactType.processor;
+	private ArtifactType determineModuleType(ModuleDefinition moduleDefinition) {
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java]: Can this method be changed to `determineArtifactType` ?

Comment (ericbottard) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java]: I don't think that would make sense, as it only applies to modules (and stream modules, for that matter)

Comment (ilayaperumalg) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java]: ok, my reasoning was that it determines and returns `ArtifactType` and I believe there is no module type now ;)

File: spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java [hunk 2]
```diff
@@ -286,37 +284,19 @@ private ModuleDefinition postProcessLibraryProperties(ModuleDefinition module) {
...
+		// Parser has already taken care of source/sink named channels, etc
+		boolean hasOutput = moduleDefinition.getParameters().containsKey(BindingProperties.OUTPUT_BINDING_KEY);
+		boolean hasInput = moduleDefinition.getParameters().containsKey(BindingProperties.INPUT_BINDING_KEY);
+		if (hasInput && hasOutput) {
+			return ArtifactType.processor;
+		} else if (hasInput) {
```
Comment (markfisher) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java]: newlines after closing braces please

File: spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java (hunk 1 above)
Comment (markfisher) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java]: The "module" types (source, processor, and sink) are still relevant as a logical type within a stream, and they are a subset of artifact types (which currently includes "task" and "library" as well). I agree with Eric that this method should be clear that it only returns module types.

**Comment by markfisher** [comment]
As part of this PR, please add a unit test to cover the case that led to the issue being reported (e.g. a stream definition such as: "queue:x > someProcessor > queue:y"). The issue is https://github.com/spring-cloud/spring-cloud-dataflow/issues/130

**Comment by whummer** [comment]
@markfisher the issue you are referring to is #128. The problem there is that the `bridge` module does not exist. Not sure if #128 is related to #130 (although it would be great to have both solved :) ). thanks

**Comment by ericbottard** [comment]
Added test, fixed formatting

**Comment by markfisher** [comment]
LGTM; merging

**Comment by markfisher** [comment]
squashed and merged as https://github.com/spring-cloud/spring-cloud-dataflow/commit/4238a95ac842627a6dc15ed9d7163164af6c93b7

**Pull request closed without merging**